src/main/java/de/tum/in/www1/artemis/service/RepositoryService.java

```diff
@@ -323,10 +321,8 @@ public boolean isClean(VcsRepositoryUrl repositoryUrl, String defaultBranch) thr
      * @return the repository if available.
      * @throws GitAPIException if the repository can't be checked out.
      * @throws IllegalAccessException if the user does not have access to the repository.
-     * @throws InterruptedException if the repository can't be checked out.
      */
-    public Repository checkoutRepositoryByName(Exercise exercise, VcsRepositoryUrl repoUrl, boolean pullOnCheckout)
-            throws IllegalAccessException, InterruptedException, GitAPIException {
+    public Repository checkoutRepositoryByName(Exercise exercise, VcsRepositoryUrl repoUrl, boolean pullOnCheckout) throws IllegalAccessException, GitAPIException {
```

Comment:
I only want to remove the `InterruptedException` being passed on from the `GitService` so that it can be removed in the places that depend on that service. I clarified that a bit more in the PR description.

Here the `IllegalAccessException` definitely has to be thrown in any case, as the permission check in line 328 throws it.